### PR TITLE
op-challenger: Add cannon-server and cannon-snapshot-freq flag

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

Add a `--cannon-server` flag to specify the binary to use as the pre-image oracle server when executing cannon.  Typically this will be `op-program`.
Add an optional `--canon-snapshot-freq` flag to control how frequently to generate snapshots when executing cannon.

**Tests**

Updated unit tests

**Metadata**

- https://linear.app/optimism/issue/CLI-4241/cli-for-cannon-inputs

### CODE REVIEW

File: op-challenger/flags/flags.go [hunk 1]
```diff
@@ -78,6 +83,12 @@ var (
 		Usage:   "L2 Address of L2 JSON-RPC endpoint to use (eth and debug namespace required)  (cannon trace type only)",
 		EnvVars: prefixEnvVars("CANNON_L2"),
 	}
+	CannonSnapshotFreqFlag = &cli.UintFlag{
+		Name:    "cannon-snapshot-freq",
+		Usage:   "Frequency of cannon snapshots to generate in VM steps (cannon trace type only)",
+		EnvVars: prefixEnvVars("CANNON_SNAPSHOT_FREQ"),
+		Value:   config.DefaultCannonSnapshotFreq,
+	}
```
Comment (refcell) [inline op-challenger/flags/flags.go:91]: Should this be frequency (the number of runs per unit of time) or instead expressed as an interval?

Frequency has a unit of cycles/unit time which isn't very expressive and confusing as opposed to interval which is just a unit of time.

**Review by refcell — COMMENTED** [review]
Small verbiage question, otherwise looks great!

File: op-challenger/flags/flags.go (hunk 1 above)
Comment (ajsutton) [reply op-challenger/flags/flags.go:91]: No it definitely has to be number of VM steps, not time as that's what cannon uses. Plus the time varies depending on what else is running on the machine so it's a lot easier to reason about how many VM steps between snapshots instead.

**Review by refcell — APPROVED** [review]

File: op-challenger/flags/flags.go (hunk 1 above)
Comment (refcell) [reply op-challenger/flags/flags.go:91]: Got it thanks for clarifying. All makes sense to me then.

**Pull request merged**